crates/ui/src/components/platform_titlebar.rs

```diff
@@ -0,0 +1,157 @@
+use gpui::{
+    div,
+    prelude::FluentBuilder,
+    px, AnyElement, Div, Element, Fill, InteractiveElement, IntoElement, MouseButton,
+    ParentElement, Render, RenderOnce, Rgba, StatefulInteractiveElement, StyleRefinement, Styled,
+    WindowAppearance::{Dark, Light, VibrantDark, VibrantLight},
+    WindowBounds, WindowContext,
+};
+use smallvec::SmallVec;
+use std::ops::Not;
+
+use crate::h_flex;
+
+#[derive(IntoElement)]
+pub struct PlatformTitlebar {
+    titlebar_bg: Option<Fill>,
+    children: SmallVec<[AnyElement; 2]>,
+    style: StyleRefinement,
+}
+
+impl Styled for PlatformTitlebar {
+    fn style(&mut self) -> &mut StyleRefinement {
+        &mut self.style
+    }
+}
+
+impl PlatformTitlebar {
+    fn render_caption_buttons(cx: &mut WindowContext) -> impl Element {
+        let btn_height = cx.titlebar_height() - cx.titlebar_top_padding();
+        let close_btn_hover_color = Rgba {
+            r: 232.0 / 255.0,
+            g: 17.0 / 255.0,
+            b: 32.0 / 255.0,
+            a: 1.0,
+        };
+
+        let btn_hover_color = match cx.appearance() {
+            Light | VibrantLight => Rgba {
+                r: 0.1,
+                g: 0.1,
+                b: 0.1,
+                a: 0.2,
+            },
+            Dark | VibrantDark => Rgba {
+                r: 0.9,
+                g: 0.9,
+                b: 0.9,
+                a: 0.1,
+            },
+        };
+
+        fn windows_caption_btn(icon_text: &'static str, hover_color: Rgba) -> Div {
+            // let mut active_color = hover_color.clone();
+            // active_color.a *= 0.2;
+            h_flex()
+                .h_full()
+                .justify_center()
+                .content_center()
+                .items_center()
+                .w(px(36.)) // TODO: get size of controls from window
+                .hover(|style| style.bg(hover_color))
+                // .active(|style| style.bg(active_color))
```

Comment:
little unsure how to get these 'active' styles to work

Reply:
I don't think the active state is relevant here.